### PR TITLE
Deprecate `TexMobject` and `TextMobject` in favour of `MathTex` and `Tex`

### PR DESCRIPTION
**Pull request by Aathish04**
## List of Changes

- Rename TextMobject to Tex
- Replace all instances of TexMobject with MathTex and TextMobject with Tex.
- Get rid of single-use TexSymbol.
- Rename SingleStringTexMobject to SingleStringMathTex.
- Add Deprecation warnings to TexMobject and TextMobject.
- Edit `__all__` for tex_mobject.py

## Motivation
The names `TextMobject` and `TexMobject` are _incredibly_ confusing to new users owing to the similarity in their names and lack of information differentiating them and the `Text` object which are wholly unrelated.

Renaming the objects to be more specific would be _very_ beneficial, especially to new users.

## Explanation for Changes
Changes include renaming the objects as suggested, and replacing all instances of the old names with the new ones.

To ease migration from 3b1b/manim, instead of outright removing `Tex[t]Mobject`, deprecation warnings have been added instead.

## Testing Status
Tests currently **fail**. I (or anyone else interested) will port the tests to the new names once #335 is merged as it refactors the testing mechanism.

## Further Comments
See discord discussion starting [here.](https://discordapp.com/channels/581738731934056449/712140817200054294/717242163465027645)

## Acknowledgement
- [x] I have read the [Contributing Guidelines](https://github.com/ManimCommunity/manim/wiki/Documentation-guidelines-(WIP))

### CODE REVIEW

File: manim/mobject/svg/tex_mobject.py [hunk 1]
```diff
@@ -26,13 +28,7 @@
 TEX_MOB_SCALE_FACTOR = 0.05
 
 
-class TexSymbol(VMobjectFromSVGPathstring):
```
Comment (leotrs) [inline manim/mobject/svg/tex_mobject.py:29]: Did this go away?

File: manim/mobject/svg/tex_mobject.py [hunk 2]
```diff
@@ -248,15 +244,15 @@ def sort_alphabetically(self):
         self.submobjects.sort(key=lambda m: m.get_tex_string())
 
 
-class TextMobject(TexMobject):
+class Tex(MathTex):
     CONFIG = {
         "alignment": "\\centering",
         "arg_separator": "",
         "type": "text",
     }
```
Comment (leotrs) [inline manim/mobject/svg/tex_mobject.py:252]: Comment (non-blocking): What never made sense to me is that I would expect MathTex() to be equivalent to something like Tex(mathmode=True) or something like that. But it seems to be the other way around, and you can't even really make one from the other. But anyway.

**Comment by leotrs** [comment]
SGTM. So the change is TexMobject -> Tex and TextMobject -> MathTex?

File: manim/mobject/svg/tex_mobject.py (hunk 1 above)
Comment (Aathish04) [reply manim/mobject/svg/tex_mobject.py:29]: Yup. It was just a renaming, and was only used in _one_ place in the same file.

**Comment by kilacoda** [comment]
I'd prefer to have Tex[t]Mobject as it is, and have `Tex` and `MathTex` as aliases, as follows:

```py
Tex = TextMobject
MathTex = TexMobject
```

**Comment by Aathish04** [comment]
> SGTM. So the change is TexMobject -> Tex and TextMobject -> MathTex?

No, exactly the other way round.

`TexMobject` -> `MathTex`, since that's what uses `LaTeX` MathMode. `TextMobject` -> `Tex` since it's just normal non-maths-y `LaTeX`.

**Comment by Aathish04** [comment]
> I'd prefer to have Tex[t]Mobject as it is, and have `Tex` and `MathTex` as aliases, as follows:
> 
> ```python
> Tex = TextMobject
> MathTex = TexMobject
> ```

Why though? I was thinking we could remove those horribly confusing names entirely, but keep them for the time being so people can migrate.

**Comment by leotrs** [comment]
If you just rename them then there is no way of issuing a deprecation warning. I agree they should go away in v2.0

**Comment by kilacoda** [comment]
Because many people are already used to these names, and understand their differences, and I don't feel it's justified to force change onto them by breaking their old code. By adding aliases, and perhaps a little one-liner note in the docstrings, we can solve the problem of confusing names without breaking peoples' existing code.

Also, `TexSymbol` looks way more intuitive while say, debugging with `print_family`, than `VMobjectFromPathString`.

Which looks better?
```console
Tex <memory_loc>
        VMobjectFromPathString <memory_loc>
        VMobjectFromPathString <memory_loc>
        VMobjectFromPathString <memory_loc>
        ...
        TexSymbol <memory_loc>
        TexSymbol <memory_loc>
        TexSymbol <memory_loc>
```

**Comment by leotrs** [comment]
This PR doesn't break code though, it just issues warnings.

I agree TexSymbol looks more convenient.

**Comment by PgBiel** [comment]
> Also, `TexSymbol` looks way more intuitive while say, debugging with `print_family`, than `VMobjectFromSVGPathString`.
it sounds like the latter is a more general name. If its only purpose is to serve Tex classes, then maybe it should receive that name, otherwise we could just subclass it with this name, as it could have other uses,

File: manim/mobject/svg/tex_mobject.py (hunk 2 above)
Comment (PgBiel) [reply manim/mobject/svg/tex_mobject.py:252]: we should probably change this at some point

**Comment by kilacoda** [comment]
> This PR doesn't break code though, it just issues warnings.

I didn't see that earlier. Apologies for that.

**Comment by leotrs** [comment]
This LGTM. Are we waiting for #335 ?

**Comment by Aathish04** [comment]
Alright, so we can all agree that the `DeprecationWarning`s are fine for now, right?

About the TexSymbol, I agree that `VMobjectFromSVGPathString` is way uglier when debugging, but it's only being used in a _single_ place, in the same file, that too.

**Comment by leotrs** [comment]
> Alright, so we can all agree that the `DeprecationWarning`s are fine for now, right?

I count four yeses. (We never decided what actually constitutes a _consensus_, but having approval from every dev who read this file is a good sign.)

> About the TexSymbol, I agree that `VMobjectFromSVGPathString` is way uglier when debugging, but it's only being used in a _single_ place, in the same file, that too.

Is TexSymbol/VMobjectFromSVGPathString going to ever be used by the end user? If so, I vote we keep it. The latter is more verbose which is useful during development, and the former is more succinct which is convenient for the user. There's nothing wrong in having an alias for convenience I think.

**Comment by Aathish04** [comment]
> > About the TexSymbol, I agree that `VMobjectFromSVGPathString` is way uglier when debugging, but it's only being used in a _single_ place, in the same file, that too.
> 
> Is TexSymbol/VMobjectFromSVGPathString going to ever be used by the end user? If so, I vote we keep it. The latter is more verbose which is useful during development, and the former is more succinct which is convenient for the user. There's nothing wrong in having an alias for convenience I think.

I think @kilacoda is in a better position to answer this, since he worked with LaTeX quite a bit when writing `Chanim`.

As for my opinion, I don't think the end user will ever use `TexSymbol` since what it does can easily be emulated by `SVGMobject` and `Tex[t]Mobject` which are more commonly used and popular anyway.

**Comment by kilacoda** [comment]
No, it's not really used by the user, rather it's for splitting up the individual parts of the SVG (`path_string`s) to be used as indexable submobjects, so it's just a name made to elaborate the fact that it's a part of a `TexMobject`.

**Comment by leotrs** [comment]
Ok in that case I'm feeling more neutral about it.

But also, it doesn't hurt to have an alias. It doesn't matter if it's used only once in the codebase if it's meant to be used for debugging. In the interest of merging this quickly, let's just leave it and move on.

**Comment by Aathish04** [comment]
Alrighty, so the only thing remaining is to port the tests for `TextMobject` and `TexMobject` over to their new names, which will be done after #335 is merged. Please review #335 if you haven't yet!

File: manim/mobject/svg/tex_mobject.py [hunk 3]
```diff
@@ -13,7 +14,8 @@
 from functools import reduce
 import operator as op
 
-from ...logger import logger
+from ... import config
+from ...config import logger
```
Comment (leotrs) [inline manim/mobject/svg/tex_mobject.py]:
```suggestion
from ... import config, logger
```

File: manim/scene/graph_scene.py [hunk 4]
```diff
@@ -412,11 +412,11 @@ def get_graph_label(
 
         Returns
         -------
-        TexMobject
+        MathTex
```
Comment (PgBiel) [inline manim/scene/graph_scene.py:415]:
```suggestion
        :class:`~.MathTex`
```

File: manim/scene/vector_space_scene.py [hunk 5]
```diff
@@ -329,8 +329,8 @@ def label_vector(self, vector, label, animate=True, **kwargs):
 
         Returns
         -------
-        TexMobject
-            The TexMobject of the label.
+        MathTex
```
Comment (PgBiel) [inline manim/scene/vector_space_scene.py:332]:
```suggestion
        :class:`~.MathTex`
```

File: manim/scene/vector_space_scene.py [hunk 6]
```diff
@@ -759,22 +759,22 @@ def add_transformable_label(
         vector : Vector
             The vector for which the label must be added.
 
-        label : TexMobject,str
-            The TexMobject/string of the label.
+        label : MathTex,str
```
Comment (PgBiel) [inline manim/scene/vector_space_scene.py:762]: let's already start enforcing the guidelines here
```suggestion
        label : Union[:class:`~.MathTex`, :class:`str`]
```

File: manim/scene/vector_space_scene.py [hunk 7]
```diff
@@ -759,22 +759,22 @@ def add_transformable_label(
...
             What the label should display after a Linear Transformation
 
         **kwargs
             Any valid keyword argument of get_vector_label
 
         Returns
         -------
-        TexMobject
-            The TexMobject of the label.
+        MathTex
```
Comment (PgBiel) [inline manim/scene/vector_space_scene.py:776]:
```suggestion
        :class:`~.MathTex`
```

File: manim/scene/vector_space_scene.py [hunk 8]
```diff
@@ -759,22 +759,22 @@ def add_transformable_label(
         vector : Vector
             The vector for which the label must be added.
 
-        label : TexMobject,str
-            The TexMobject/string of the label.
+        label : MathTex,str
+            The MathTex/string of the label.
 
-        transformation_name : str, TexMobject, optional
+        transformation_name : str, MathTex, optional
```
Comment (PgBiel) [inline manim/scene/vector_space_scene.py:765]:
```suggestion
        transformation_name : Union[:class:`str`, :class:`~.MathTex`], optional
```

File: manim/scene/vector_space_scene.py [hunk 9]
```diff
@@ -759,22 +759,22 @@ def add_transformable_label(
...
             The name to give the transformation as a label.
 
-        new_label : TexMobject,str, optional
+        new_label : MathTex,str, optional
```
Comment (PgBiel) [inline manim/scene/vector_space_scene.py:768]:
```suggestion
        new_label : Union[:class:`str`, :class:`~.MathTex`], optional
```

File: manim/scene/vector_space_scene.py [hunk 10]
```diff
@@ -799,7 +799,7 @@ def add_title(self, title, scale_factor=1.5, animate=False):
 
         Parameters
         ----------
-        title : str,TexMobject,TextMobject
+        title : str,MathTex,Tex
```
Comment (PgBiel) [inline manim/scene/vector_space_scene.py:802]:
```suggestion
        title : Union[:class:`str`, :class:`~.MathTex`, :class:`~.Tex`]
```

**Comment by Aathish04** [comment]
Unmarking as draft, since the `Tex[T]Mobject` tests don't need to be ported as they were removed in #335 .

The only test that fails now is the formatting one, which should be fixed in due course I think.

**Comment by Aathish04** [comment]
I've run the new, updated version of black on the files that have been touched by this PR, as per @leotrs 's suggestion.

**Pull request merged**